interfaces/builtin/gpio.go

```diff
@@ -122,8 +122,8 @@ func (iface *GpioInterface) ConnectedSlotRichSnippet(plug *interfaces.Plug, slot
 				serviceName: systemd.Service{
 					Type:            "oneshot",
 					RemainAfterExit: true,
-					ExecStart:       fmt.Sprintf("/bin/sh -c 'echo %d > /sys/class/gpio/export'", gpioNum),
-					ExecStop:        fmt.Sprintf("/bin/sh -c 'echo %d > /sys/class/gpio/unexport'", gpioNum),
+					ExecStart:       fmt.Sprintf("/bin/sh -c 'test ! -e /sys/class/gpio/gpio%d && echo %d > /sys/class/gpio/export'", gpioNum, gpioNum),
```

Comment:
Doing an "echo 24 > /sys/class/gpio/export" with the GPIO already being exported gives 

```
root@localhost:~# ls -alh /sys/class/gpio/gpio24
lrwxrwxrwx 1 root root 0 Oct 31 10:45 /sys/class/gpio/gpio24 -> ../../devices/platform/soc/3f200000.gpio/gpio/gpio24
root@localhost:~# echo 24 > /sys/class/gpio/export
-bash: echo: write error: Device or resource busy
```

Which would lead to a failure of the service unit. Think about scenarios where people are manually playing with GPIO's and then installing snaps dealing with them as well. But you are right the logic is wrong here. For start it should be actually

```
test -e /sys/class/gpio/gpio24 || echo 24 > /sys/class/gpio/export
```